src/components/molecules/movie-reaction-panel/reaction-rate.tsx

```diff
@@ -19,56 +19,66 @@ const TypedRiStarFill = createTypedIcon(RiStarFill);
 export const ReactionRate = ({
   wantToSee,
   movieId,
+  hasSeen,
 }: {
-  wantToSee?: number;
+  wantToSee?: number | null;
   movieId?: string;
+  hasSeen?: boolean | null;
 }) => {
-  //TODO replace with useOptimistic
-  const [pending, setPending] = useState(false);
+  const [isPending, startTransition] = useTransition();
   const emptyStarPressed = wantToSee === 0;
   const halfStarPressed = wantToSee === 1;
   const fullStarPressed = wantToSee === 2;
-
-  const handleReaction = async (reaction: number) => {
-    if (reaction === wantToSee) return;
-
-    setPending(true);
-    try {
-      movieId && (await addOrUpdateReaction(movieId, reaction));
-    } finally {
-      setTimeout(() => {
-        setPending(false);
-      }, 1000);
-    }
+  const handleReaction = async (
```

Comment:
może czytelniej by było jakby przyjmować tu obiekt `{ wantToSeeReaction: number; hasSeenReaction: boolean }` , wtedy nie trzeba by przekazywać na dole !!hasSeen, tylko to co chcesz updatować